openforcefields/tests/test_water_models.py

```diff
@@ -160,7 +211,7 @@ def test_tip5p(water_molecule):
 
 @pytest.mark.parametrize(
     "water_model,pattern",
-    [("tip3p", "^tip3p(?!.*fb)"), ("tip3p_fb", "^tip3p_fb"), ("opc", "^opc")],
+    [("tip3p", "^tip3p(?!.*fb)"), ("tip3p_fb", "^tip3p_fb"), ("tip4p_fb", "^tip4p_fb"), ("opc3", "^opc"), ("opc", "^opc(?!.*3)")],
```

Comment:
I could use a second opinion on this regex to make sure that `"^opc"` doesn't hit "opc3".

Reply:
I think works for the files that currently exist

```python
In [1]: from openff.toolkit.typing.engines.smirnoff.forcefield import (
   ...:         get_available_force_fields,
   ...:     )

In [2]: import re

In [3]: pattern = "^opc(?!.*3)"

In [4]: list(
   ...:         filter(lambda x: re.match(pattern, x) is not None, get_available_force_fields())
   ...:     )
/Users/mattthompson/mambaforge/envs/interchange-examples-env/lib/python3.9/site-packages/smirnoff99frosst/smirnoff99frosst.py:11: UserWarning: Module openff was already imported from None, but /Users/mattthompson/software/openff-interchange is being added to sys.path
  from pkg_resources import resource_filename
Out[4]: ['opc-1.0.0.offxml', 'opc.offxml']
```

though I have one suggestion to consider if I'm not misunderstanding anything. `^opc(?!3)` might be safer against the possibility of something like this, since this lookahead only looks at the character immediately following `opc`

```python
In [1]: import re

In [2]: re.match("^opc(?!3)", "opc_1.3.0.offxml")
Out[2]: <re.Match object; span=(0, 3), match='opc'>  # I think this result is desired

In [3]: re.match("^opc(?!.*3)", "opc_1.3.0.offxml")
# I think this result is NOT desired

In [4]: re.match("^opc(?!.*3)", "opc3_1.3.0.offxml")
# I think this result is desired

In [5]: re.match("^opc(?!3)", "opc3_1.3.0.offxml")
# I think this result is desired
```

I'm very much learning regex as a I go, so please do validate my work yourself